support/cas-server-support-ldap/src/test/java/org/apereo/cas/authentication/ActiveDirectoryJndiSamAccountNameLdapAuthenticationHandlerTests.java

```diff
@@ -0,0 +1,51 @@
+package org.apereo.cas.authentication;
+
+import org.apereo.cas.util.junit.EnabledIfContinuousIntegration;
+import org.springframework.test.context.TestPropertySource;
+
+/**
+ * Unit test for {@link LdapAuthenticationHandler}.
+ * This test demonstrates using the AD type with the JNDI Provider.
+ * Using the Ldaptive UnboundID provider would fail because the name doesn't validate as a DN.
+ * Login name is sAMAccountName (aka "short" windows login name), bind as DOMAIN\USERNAME.
+ * This allows for a dnFormat that the UnboundID provider would not allow since it is not a valid DN.
+ * Issues:
+ *  - This configuration doesn't retrieve any attributes as part of the authentication.
+ *  - If the pool passivator is CLOSE, and the both success and failure tests run, the failure test will
+ *      fail with wrong exception type. If multiple success tests are added, every other test will fail.
+ *  - No ssl or startTls due to JDK-8217606 which currently impacts JndiProvider
+ * @author Hal Deadman
```

Comment:
I strongly recommend, though no big deal at all, to use a `<code>` tag whenever you can. Things such as `sAMAccountName`, `CLOSE` are good candidates. Also, `JndiProvider` should be using a link tag?